lib/IRGen/IRGenModule.cpp

```diff
@@ -1166,7 +1166,9 @@ void IRGenModule::emitAutolinkInfo() {
     }
     auto EntriesConstant = llvm::ConstantDataArray::getString(
         LLVMContext, EntriesString, /*AddNull=*/false);
-
+    if (TargetInfo.OutputObjectFormat == llvm::Triple::ELF)
```

Comment:
I think it would be nice to add a small comment above this explaining that `0x80000000` is `SHF_EXCLUDE` and that this ensures the section is removed by the linker from the final binary.  It would also be a good idea to explain why this is being done by means of module level inline assembly, since it may not be immediately apparent that we cannot set MC level flags here.